### ISSUE TITLE
Fix `media-feature-name-no-unknown` false positives for namespaced dollar variables and range context queries

### ISSUE THREAD

**Issue by ArianeBouchardConformit** [stylelint/stylelint#8818]
### What minimal example or steps are needed to reproduce the bug?

media-query-sizes.scss

```scss
$data-table-view-create-button-compact: 600px;
```

CreateButton.vue

```scss
@use '~@/assets/sass/base/measurements/media-query-sizes.scss';

.create-button{
  &__button {
    &--view-type-full {
      @media (max-width: media-query-sizes.$data-table-view-create-button-compact) {
        display: none;
      }
    }
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
   "extends": [ "stylelint-config-standard-scss", "stylelint-config-standard-vue/scss" ]
}
```

### How did you run Stylelint?

Can't do a demo because it requires several style files.

config:
```javascript
const STYLELINT_CONFIG = {
   extends: [ 'stylelint-config-standard-scss', 'stylelint-config-standard-vue/scss' ],
}

export default STYLELINT_CONFIG
```

### Which Stylelint-related dependencies are you using?

```json
      "stylelint": "^16.20.0",
      "stylelint-config-standard-scss": "^15.0.1",
      "stylelint-config-standard-vue": "^1.0.0",
      "stylelint-scss": "^6.12.0",
```

### What did you expect to happen?

The media query passes fine, either right in Stylelint, either thanks to something stylelint-scss is doing.

### What actually happened?

Warning:

```
Unexpected unknown media feature name "media-query-sizes"  media-feature-name-no-unknown
```

By the way, using #{} variable interpolation works fine. I just feel like it should probably not be required?

**Comment by jeddy3** [comment]
@ArianeBouchardConformit Thanks for the report and for using the template.

It seems `media-query-sizes` is being incorrectly parsed as a feature name rather than part of the feature value. However, I can't reproduce the problem as a [SCSS file](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0YBOMCGAXGBaARgK5ZYQB2wAOmQAQ0BkA+o0SeTVbXQ7rgG4BLGAHdcWAJ4AHPADNCAG3kdq3bgAEAtjCgCMNABQaMAD1zCBULAAt4NLToy4AjoRipxuAM4CAXjE9IACRQ2I5YGPjyeIIiuGiYOATEpGRxEBqSGGBYAJTKXKp0Op6S8hjitmTkMADcKoUAvvV0TVytDSAANCAyAlEAchhaCCAwxkOlMEieYJ6eXeDkvQDmI5x0lKPGOGRQnpu2ANrNNJueElHyAmRYaWQrXuG7GKhQXrP7IPUAutQd3ZB7gJlgAxCCoIxYEYAK085AWsEk80Q61OIHO4ku1ywBzRZRw502nXqZwuMCuNzuD3OGGer3ec1xm3x-hxXzIHQaQA) or within a [Vue one](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnALgTwDYwARYEMA7AcwF4AdEFMFFKgPgqIDowAnGAtGAWgCMArmjQQiwZrlwAyAPqyhIsbglEpU6b14A3AJYwA7r0wAHPgDNBWLCsnqpAAQC2MKLoK4AFE4IAPXga6UGgAFvC4Lm4EvACOgjDsGLwougBeMCgsACRQ3NFoBPw4OvpGHFw8AsKiRLyQTiYEYGgAlLZq9upuKCaEGOFEYjAA3Hb2AL5juJMdMzNIAPTo2DAMIAA0IOa6OAByBC4IIDC+B70wLNrxG+Bi2yRHqlJUJzxEUPQIuADaU1TLOCwuiIaDqd10JGSBXeBHYUGStE+6z+1EwgOBoMgRHuUOIuThOniS0RVDsAF1mOMblj7gAxCDsHxoI4AKxQYhusBMKEedn+aJgQJBVHCVEIPHQVGRHX5KyFmPBkPQeNh8JodBFuDF3AyaClfNRcoxYOxENxMIJVxgmu1Er1IEpIHGQA).

Can you provide another a demo or a repo with a minimum reproducible example to help us resolve the issue, please? 

> Can't do a demo because it requires several style files.

Stylelint does static analysis on single files and doesn't resolve imports, so it should be possible to create a standalone example to demostrate the bug.

**Comment by ArianeBouchardConformit** [comment]
Sorry, my example wasn't right. This is the case that causes the issue:

```scss
@use '@/assets/sass/base/measurements/media-query-sizes.scss';

.create-button{
  &__button {
    &--view-type-full {
      @media (width <= media-query-sizes.$data-table-view-create-button-compact) {
        display: none;
      }
    }
  }
}
```

https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBXAzjABAciQegEM0MAXNPNI8gIyJjwFsYiUAnGJgOzMZigEsCAWgCOKGKwCeQtPwBeMNADo0YYtgDcAHU46lYdgRIwh1FCRIROwHZkwAyAPqOzFq5huc7d+0KEA3fhgAdyESSQAHEwAzFAAbOI9bbzskJgECTAAKYP4oEgALTAAeAF5MdMFRcSkZeUUlABIoI2ESAmo4k0CQoQNmY1NzS04+iAYIgjASAEokrxTvATQIuIJJeExOKxhtBZSAX2S7I4XTg5AAGhBo-i6AOQImBBAYAA8n1ZgVNTQr8CstwA5i9PHYtK83sZOFA0BDNgBtTAQtDhLpxfjcMacYEydowgisKAyX4Qy7IkCoyTozEkbG41EEAlEgLiCikkCYAC6l2SENY8UU8Pmi1FFNUxDwcQgBGJk1YJEEcSE72hsiswqREIIcWC6zhnK553+kBx-CBADEIKwGEYXgArNBWf6wCJ-RBglFomAY7jCiEAPQAjAA2JQAJgADEpI2Tjt4vdSfbT6ea8UyWiyJQbNoGgwBWGNKINx-YJyne31000M-GZ4n+cT+kDBouxq7x8EVpNVknEZsBsNB8MxiE6C4HIA

**Comment by jeddy3** [comment]
@ArianeBouchardConformit Thanks for the reproducible example.

I can replicate the bug with:

```scss
@media (width <= foo.$bar) {}
```

```json
{
  "rules": {
    "media-feature-name-no-unknown": true
  }
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbGUCWBDABACgHcMoAXACxwB4BeHAOwGcA6AEgDcsAnASh2AF8QAGhAAzDABsYAOSxoEIGAA85ABylMGYBg2HgIdcQHMFwADp0cOMyE4BXKQxvw+Fq1ZtpMWALSiYWCR2nDA+dHKhdBA+dnQA1lEEdM44JPYwbjj8FoIikIYYRgBiEJwogQoAVgwGerCquojmljYMJACeUhIYdCQpNgB6AIwAbEwATAAMTJM2OSD8QA).

It seems to be the combination of range-context queries and namedspaced dollar variables.

The rule should ignore these.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8890
